hydra-node/src/Hydra/Snapshot.hs

```diff
@@ -38,6 +38,9 @@ instance (Arbitrary tx, Arbitrary (UTxOType tx)) => Arbitrary (Snapshot tx) wher
     , confirmed' <- shrink (confirmed s)
     ]
 
+genSnapShot :: Gen SnapshotNumber
+genSnapShot = arbitrary
```

Comment:
This is weird, but acceptable if needed. Maybe we can get rid of overlapping instances by not importing the `Natural` orphans.